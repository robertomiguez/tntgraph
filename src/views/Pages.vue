<template>
  <div id="main">
    <ul>
      <li v-for="(page, i) in pages" :key="i">
        <Card 
          :title="page.title" 
          :img="page.img" 
          :description="page.description"  
          :to="page.to"
        />
      </li>
    </ul>
  </div>
</template>

<script>
import Card from "../components/Card"

export default {
  name: `Pages`,
  components: {
    Card
  },  
  data: function() {
    return {
      pages: []
    }
  },
  mounted: function() {
    this.loadPages()
  },
  methods: {
    loadPages: function () {
      try {
        this.pages = 
        [
          {
            title: `Homicides in Brazil`,
            img: require(`@/assets/homicides.png`),
            description: `This is a scary statistics but sadly a true one`,
            to: `Homicides`            
          },
          {
            title: `Covid 19 - United States`,
            img: require(`@/assets/graph.png`),
            description: `Numbers of Covid 19 in United States`,
            to: `Covid19`            
          },
          {
            title: `Example`,
            img: require(`@/assets/example.png`),
            description: `this is an example...`,
            to: `Pages`            
          },
          {
            title: `Example`,
            img: require(`@/assets/example.png`),
            description: `this is an example...`,
            to: `Pages`            
          },
          {
            title: `Example`,
            img: require(`@/assets/example.png`),
            description: `this is an example...`,
            to: `Pages`            
          },
          {
            title: `Example`,
            img: require(`@/assets/example.png`),
            description: `this is an example...`,
            to: `Pages`            
          },
          {
            title: `Example`,
            img: require(`@/assets/example.png`),
            description: `this is an example...`,
            to: `Pages`            
          }
        ]
      } catch (error) {
        console.log(error)
      }
    }
  }
}
</script>

<style scoped>
#main {
  background-color: rgb(224, 223, 223);
  left: 0;
  right: 0;
  position: fixed;
}
h1 {  
    text-align: center; 
}
ul {
    list-style-type: none;
    margin: 0;
    padding: 0px;
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
    margin-top: 80px;
}
li {
    width: 350px;
    background: #FFF;
    border-radius: 15px;
    padding: 15px;
    text-align: center;
    margin: 5px;
}
</style>
