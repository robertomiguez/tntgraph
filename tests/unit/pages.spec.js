import { shallowMount } from '@vue/test-utils'
import Pages from '@/views/Pages.vue'

let wrapper

beforeEach(() => {
  wrapper = shallowMount(Pages, {
    propsData: {},
    mocks: {},
    stubs: {},
    methods: {}
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe(`Pages`, () => {
  test(`regression test`, () => {
    expect(wrapper.element)
      .toMatchSnapshot()
  })  

})