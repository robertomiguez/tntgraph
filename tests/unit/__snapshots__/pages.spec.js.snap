// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages regression test 1`] = `
<div
  id="main"
>
  <ul>
    <li>
      <card-stub
        description="This is a scary statistics but sadly a true one"
        img=""
        title="Homicides in Brazil"
        to="Homicides"
      />
    </li>
    <li>
      <card-stub
        description="Numbers of Covid 19 in United States"
        img=""
        title="Covid 19 - United States"
        to="Covid19"
      />
    </li>
    <li>
      <card-stub
        description="this is an example..."
        img=""
        title="Example"
        to="Pages"
      />
    </li>
    <li>
      <card-stub
        description="this is an example..."
        img=""
        title="Example"
        to="Pages"
      />
    </li>
    <li>
      <card-stub
        description="this is an example..."
        img=""
        title="Example"
        to="Pages"
      />
    </li>
    <li>
      <card-stub
        description="this is an example..."
        img=""
        title="Example"
        to="Pages"
      />
    </li>
    <li>
      <card-stub
        description="this is an example..."
        img=""
        title="Example"
        to="Pages"
      />
    </li>
  </ul>
</div>
`;
